### ISSUE TITLE
Problem with Unicode in dal ver.3

### ISSUE THREAD

**Issue by epanchee** [yourlabs/django-autocomplete-light#592]
I try to use your dal v3 and found that  it has lack of documentation.
I tried your live example and I works good even with non-unicode symbols.
My simple app interacts with db and stores data in Russian.
So, there is the problem that I was solving for 5 days due lack of information about dal v3.
I implemented form, view and added url accoring new documentation. It seems that this side works properly (I inspected it via F12 tool, ajax requests work).
Here is the full stack that I got when trying to write some russian letters:
```
Internal Server Error: /admin/visitor-autocomplete/
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\django\core\handlers\base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "C:\Python27\lib\site-packages\django\core\handlers\base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Python27\lib\site-packages\django\views\generic\base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "C:\Python27\lib\site-packages\django\views\generic\base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "C:\Python27\lib\site-packages\dal\views.py", line 23, in get
    return super(ViewMixin, self).get(request, *args, **kwargs)
  File "C:\Python27\lib\site-packages\django\views\generic\list.py", line 175, in get
    return self.render_to_response(context)
  File "C:\Python27\lib\site-packages\dal_select2\views.py", line 35, in render_to_response
    'results': self.get_results(context) + create_option,
  File "C:\Python27\lib\site-packages\dal_select2\views.py", line 19, in get_results
    } for result in context['object_list']
  File "C:\Python27\lib\site-packages\dal\views.py", line 42, in get_result_label
    return six.text_type(result)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xd0 in position 0: ordinal not in range(128)
```
I tried to encode/decode to/from utf-8. It didn't help. After that I opened function `get_result_label` and added `return six.text_type(str(result).decode('utf-8'))`. It helped me out. But it seems very weird. Like workaround.

Could you advice me how to get rid of this problem or you would make changes for that?
Thank you in advance!

**Comment by jpic** [comment]
Thanks for your report, we should have a patch and a test case indeed, but I'm curious why it works on the test project and not for you, are you running mysql perhaps ? what version of python and django ?

**Comment by epanchee** [comment]
I confused as well. Atm I use SQLite, python 2.7.11, django 1.9.1.

**Comment by jpic** [comment]
Can you reproduce it by running the test project locally ?

**Comment by jpic** [comment]
What version of sqlite ?

**Comment by epanchee** [comment]
Just tryed to reproduce this error on your test project locally, but it works same as on deployed test project. It's confusing me more. So, what's wrong with my code?

**Comment by epanchee** [comment]
I found why this problem occurs. It was my mistake, not yours. For my model I used self.somefield.encode('utf-8') to represent it in django admin. That's why I had to change `get_result_label` function to decode it back from utf-8. The solution is to use __unicode__ insted __str__. 
Actually I'm newbie in python and using python2, but I thought that right 'to string' function is __str__.

Sorry for the inconvenience.
